modules/mtl/arrow-mtl/src/main/kotlin/arrow/mtl/extensions/kleisli.kt

```diff
@@ -249,3 +254,30 @@ interface KleisliEqK<F, D> : EqK<KleisliPartialOf<F, D>> {
       }
     }
 }
+
+@extension
+interface KleisliMonadPlus<F, D> : MonadPlus<KleisliPartialOf<F, D>>, KleisliMonad<F, D>, KleisliAlternative<F, D> {
+  override fun MF(): Monad<F>
+  override fun AL(): Alternative<F>
+  override fun AF(): Applicative<F> = AL()
+}
+
+@extension
+interface KleisliMonadLogic<F, D> : MonadLogic<KleisliPartialOf<F, D>>, KleisliMonadPlus<F, D> {
+  fun ML(): MonadLogic<F>
+  override fun MF(): Monad<F> = ML()
+  override fun AL(): Alternative<F> = ML()
+
+  override fun <A> Kind<KleisliPartialOf<F, D>, A>.splitM(): Kind<KleisliPartialOf<F, D>, Option<Tuple2<Kind<KleisliPartialOf<F, D>, A>, A>>> =
+    this.fix().let { fa ->
+      Kleisli(
+        AndThen(fa.run).andThen {
+          ML().run {
+            it.splitM().flatMap { option ->
+              option.fold({ just(Option.empty<Tuple2<Kind<KleisliPartialOf<F, D>, A>, A>>()) }, { (fa, a) -> just(Option.just(Kleisli.liftF<F, D, A>(fa) toT a)) })
```

Comment:
Please avoid choosing colliding names, makes this confusing to read

Reply:
yes